src/next-version-pr.ts

```diff
@@ -0,0 +1,88 @@
+import { cdk } from "projen";
+import { GithubWorkflow } from "projen/lib/github";
+import { JobPermission } from "projen/lib/github/workflows-model";
+
+const PULL_REQUEST_REF = "${{ github.event.pull_request.head.ref }}";
+const PULL_REQUEST_REPOSITORY =
+  "${{ github.event.pull_request.head.repo.full_name }}";
+
+export class NextVersionPr {
+  constructor(project: cdk.JsiiProject, githubSecret: string) {
+    const workflow = new GithubWorkflow(
+      project.github!,
+      "next-cdktf-version-pr"
+    );
+    workflow.on({ workflowDispatch: {} });
+    workflow.addJob("pr-against-next", {
+      runsOn: ["ubuntu-latest"],
+      permissions: {
+        contents: JobPermission.WRITE,
+        pullRequests: JobPermission.WRITE,
+      },
+      env: {
+        CI: "true",
+        CHECKPOINT_DISABLE: "1",
+      },
+      steps: [
+        {
+          name: "Checkout",
+          uses: "actions/checkout@v3",
+          with: {
+            ref: PULL_REQUEST_REF,
+            repository: PULL_REQUEST_REPOSITORY,
+          },
+        },
+        {
+          name: "Remove old PR",
+          run: [
+            `PR_NUMBER_TO_CLOSE=$(gh pr list | grep "cdktf-next-pr" | awk '{ print $1 }')`,
+            `if [ -z "$PR_NUMBER_TO_CLOSE" ]; then`,
+            `  echo "No PR to close"`,
+            `else`,
+            `  gh pr close $PR_NUMBER_TO_CLOSE`,
+            `fi`,
+          ].join("\n"),
+          env: {
+            GH_TOKEN: githubSecret,
+          },
+        },
+        {
+          name: "Install",
+          run: "yarn install",
+        },
+        {
+          name: "Upgrade CDKTF",
+          // Replace 'cdktfVersion: "0.12.0"' with 'cdktfVersion: "<next>"'
+          run: [
+            // We need this to get the full tag of the @next version
+            `yarn upgrade cdktf@next`,
+            `CDKTF_VERSION=$(jq .devDependencies.cdktf -r < package.json)`,
+            `sed -i "s/cdktfVersion: \".*\",/cdktfVersion: \\"$CDKTF_VERSION\\",/" .projenrc.js`,
+            `cat .projenrc.js`,
+          ].join("\n"),
+        },
+        {
+          name: "Run projen",
+          run: "yarn run upgrade",
+        },
+        {
+          name: "Regenerate bindings",
+          run: "npx cdktf get",
```

Comment:
I think the proper command to update bindings would be `yarn fetch` (it does [a bit more than get](https://github.com/hashicorp/cdktf-provider-aws/blob/e6280ad57d36c3488acab3e53e4b6df6bcd15d21/.projen/tasks.json#L134)).

Reply:
The other `upgrade-provider` workflow [also runs `compile` and `docs`](https://github.com/hashicorp/cdktf-provider-aws/blob/e6280ad57d36c3488acab3e53e4b6df6bcd15d21/.github/workflows/provider-upgrade.yml#L24-L25). But I think we don't need them as the purpose of this PR is to possibly fail (which will happen at compile probably) 🤔

Reply:
I had issues where fetch didn't work, but I can try it again

Reply:
maybe it is good enough to run get, as e.g. copying saving the version to a json file is not necessary as we will never merge the test PR.
I'd still prefer fetch though 😁